### ISSUE TITLE
Bracket Pair Colorizer is no longer being maintained.

### ISSUE THREAD

**Issue by czzonet** [CoenraadS/Bracket-Pair-Colorizer-2#475]
Please include the language you are using, a screenshot, and copy/pastable example.

![Snipaste_2021-12-14_15-37-35](https://user-images.githubusercontent.com/16220009/145954116-c8c987c1-8415-4714-b4bd-98e1e8fc76d7.png)

**Comment by czzonet** [comment]
Oh, it has become a native feature in VSCode.

**Comment by czzonet** [comment]
![image](https://user-images.githubusercontent.com/16220009/145954876-56debfa0-c3c8-4b36-b652-19fe87b82b6f.png)

**Comment by aminomancer** [comment]
Unfortunately VS code's indentation guides `editor.guides.bracketPairs` can't draw in the gutter like the extension's can. So indent level 0's vertical guides are just invisible. Hopefully microsoft fixes it this soon

**Comment by deitry** [comment]
Bracket Pair Colorier:
![image](https://user-images.githubusercontent.com/6714840/145979933-df66a683-2a9d-4da3-a87e-e0e57d19333d.png)
Native:
![image](https://user-images.githubusercontent.com/6714840/145979943-5bfd9886-c31a-4b94-827b-b38c1849e2b0.png)

I found visual style provided by extension much more appealing, so I will continue to use this extension.

It would be great if there would be an option to suppress "no longer maintained" message.

**Comment by nicklasbns** [comment]
it seems that the native implementation does not support custom colors and will only let you use the 3 default colors, so therefore i will also continue using the extension

**Comment by Diogo-Rossi** [comment]
It seems the native implementation does not put **horizontal line** in pairs.
Does anyone have the same? Is there some missing setting on the native?

## Bracket Pair Colorizer:
![Animation0](https://user-images.githubusercontent.com/57204023/146001773-93bed9f6-6f96-4b89-9856-30eebee22d4f.gif)

## Native:
![Animation1](https://user-images.githubusercontent.com/57204023/146001782-980c04f0-9b8e-401b-a0f8-4a2ed34b928a.gif)

**Comment by Jimbo99** [comment]
> It seems the native implementation does not put **horizontal line** in pairs. Does anyone have the same? Is there some missing setting on the native?
> 
> ## Bracket Pair Colorizer:
> ![Animation0](https://user-images.githubusercontent.com/57204023/146001773-93bed9f6-6f96-4b89-9856-30eebee22d4f.gif)
> 
> ## Native:
> ![Animation1](https://user-images.githubusercontent.com/57204023/146001782-980c04f0-9b8e-401b-a0f8-4a2ed34b928a.gif)

I think I read horizontal guides are on Microsoft's "to do" list, but I'm not sure where I saw that.

**Comment by NickAtTyl** [comment]
![image](https://user-images.githubusercontent.com/86842846/146005393-c1136ecc-2da2-45f2-8d62-e72e91d10ef3.png)

@Diogo-Rossi There is a setting for horizontal line pairing but it doesn't seem to work for me.

**Comment by Jimbo99** [comment]
> @Diogo-Rossi There is a setting for horizontal line pairing but it doesn't seem to work for me.

I didn't know that, thank you. I'm actually still using Bracket Colorizer 2, since even the vertical guides of the inbuilt VSCode colorizer don't look as good.

**Comment by silkfire** [comment]
> Bracket Pair Colorizer: ![image](https://user-images.githubusercontent.com/6714840/145979933-df66a683-2a9d-4da3-a87e-e0e57d19333d.png) Native: ![image](https://user-images.githubusercontent.com/6714840/145979943-5bfd9886-c31a-4b94-827b-b38c1849e2b0.png)
> 
> I found visual style provided by extension much more appealing, so I will continue to use this extension.
> 
> It would be great if there would be an option to suppress "no longer maintained" message.

Just click on the **Don't show again** button to hide the notification.

**Comment by leochoo** [comment]
@CoenraadS Thank you for your awesome work! So did VSCode reach out to you to put this as a native feature or sth?

**Comment by Y7n05h** [comment]
@CoenraadS Thank you for your great work．
I will still use this plugin until it doesn't work properly or Microsoft implements it better.

**Comment by nicbet** [comment]
Thanks for all your hard work and contributions to making the lives of hundreds of thousands of devs better over the last 5 years.

Like others, I've tried the native extension and it can't touch the functionality of your extension even if the underlying engine is the same. So like others I will continue to use this "unmaintained" extension until Microsoft (if ever) brings their native functionality up to par.

Wishing you wonderful Holidays from 🇨🇦

**Comment by asolux** [comment]
@CoenraadS Also a big thank you from me. I can only agree with @Y7n05h comment and will continue to use the plugin as well.

**Comment by aminomancer** [comment]
> it seems that the native implementation does not support custom colors and will only let you use the 3 default colors, so therefore i will also continue using the extension

> It seems the native implementation does not put horizontal line in pairs.

That's not true at all. You need to read the [October](https://code.visualstudio.com/updates/v1_62#_improved-bracket-pair-guides) and [September](https://code.visualstudio.com/updates/v1_61#_bracket-pair-guides) release notes.

    "editor.guides.bracketPairs": "active",
    "editor.guides.bracketPairsHorizontal": "active",
    "editor.guides.indentation": true,
    "editor.guides.highlightActiveIndentation": true,
    "editor.bracketPairColorization.enabled": true,
    "workbench.colorCustomizations": {
        "editorBracketHighlight.foreground1": "#ffffff",
        "editorBracketHighlight.foreground2": "#ffffff",
        "editorBracketHighlight.foreground3": "#ffffff",
        "editorBracketHighlight.foreground4": "#ffffff",
        "editorBracketHighlight.foreground5": "#ffffff",
        "editorBracketHighlight.foreground6": "#ffffff",
        "editorBracketHighlight.unexpectedBracket.foreground": "#ffffff",
        "editorBracketPairGuide.background1": "#ffffff",
        "editorBracketPairGuide.background2": "#ffffff",
        "editorBracketPairGuide.background3": "#ffffff",
        "editorBracketPairGuide.background4": "#ffffff",
        "editorBracketPairGuide.background5": "#ffffff",
        "editorBracketPairGuide.background6": "#ffffff",
        "editorBracketPairGuide.activeBackground1": "#ffffff",
        "editorBracketPairGuide.activeBackground2": "#ffffff",
        "editorBracketPairGuide.activeBackground3": "#ffffff",
        "editorBracketPairGuide.activeBackground4": "#ffffff",
        "editorBracketPairGuide.activeBackground5": "#ffffff",
        "editorBracketPairGuide.activeBackground6": "#ffffff",
    }

Many of these comments are outright wrong. Like I said before, the only deficit in vs code's implementation is the inability to draw in the gutter. Hopefully this will be fixed soon. I have already opened [an issue](https://github.com/microsoft/vscode/issues/139067) for this and it's been assigned. Keep in mind that vs code's implementation is vastly faster and more performant, and does not suffer from the bug where bracket pair guides get crooked due to line indentation changing. That is why this extension has been deprecated. It's great and I'll keep using it until the gutter issue is fixed, but vs code's implementation is faster and isn't missing anything else.

**Comment by Rados51** [comment]
I am keeping this extension also. Native implementation work like .... in my case and I don't have time to play with all the settings to keep the same look and functionality.

**Comment by aminomancer** [comment]
> I am keeping this extension also. Native implementation work like .... in my case and I don't have time to play with all the settings to keep the same look and functionality.

The default color settings are almost identical. You only need to change a few settings to enable the feature. Everything else required modifying the extension's settings in the first place. You can just copy paste this into your settings.json file, or search for them in the settings UI.
```
"editor.guides.bracketPairs": true,
"editor.guides.bracketPairsHorizontal": "active",
"editor.guides.highlightActiveIndentation": true,
"editor.bracketPairColorization.enabled": true,
```

**Comment by nicklasbns** [comment]
> That's not true at all. It supports the same custom colors as the extension and supports up to 6 unique levels, and it does have horizontal bracket pair guides.

Oh, thank you. I did not find these settings when i was exploring it at release. I might then switch over at some point. Though i would like to have more than 6 colors, since i have 11 in my current setup...

**Comment by aminomancer** [comment]
Yeah some of the articles that mention the new feature don't mention all the settings. I think because they weren't all added at once.

Personally I use 5 colors, I try to keep the contrast between adjacent colors as high as possible and it's hard to do that with more than 5 or 6. I think that's the reasoning behind the maximum of 6. I saw someone who set it up like a literal rainbow but it seems not very functional, since the whole point is distinguishing one bracket pair from the others nearby. Even 6 colors seemed excessive, since the first and last are also adjacent when the pattern restarts. So 5 was the sweet spot for me. But if you need 11 you can always post an issue on the vscode repo.

**Comment by ferrolho** [comment]
Has anyone figured out the best colour scheme settings for the native feature to look like the default settings of this extension? And if yes, could you share it please? :D

**Comment by rareyman** [comment]
> Many of these comments are outright wrong. Like I said before, the only deficit in vs code's implementation is the inability to draw in the gutter. Hopefully this will be fixed soon. I have already opened [an issue](https://github.com/microsoft/vscode/issues/139067) for this and it's been assigned. Keep in mind that vs code's implementation is vastly faster and more performant, and does not suffer from the bug where bracket pair guides get crooked due to line indentation changing. That is why this extension has been deprecated. It's great and I'll keep using it until the gutter issue is fixed, but vs code's implementation is faster and isn't missing anything else.

I can confirm that I was able to re-implement the colors I used in `bracket-pair-colorizer-2` pretty easily using the code @aminomancer posted. Here is my colors based on the Nord color theme:

```
…
    “editorBracketHighlight.foreground1": "#39b6b4",
    "editorBracketHighlight.foreground2": "#c0616a",
    "editorBracketHighlight.foreground3": "#d08871",
    "editorBracketHighlight.foreground4": "#a3be8c",
    "editorBracketHighlight.foreground5": "#b48fad",
    "editorBracketHighlight.foreground6": "#5f7083",
    "editorBracketPairGuide.activeBackground1": "#39b6b4",
    "editorBracketPairGuide.activeBackground2": "#c0616a",
    "editorBracketPairGuide.activeBackground3": "#d08871",
    "editorBracketPairGuide.activeBackground4": "#a3be8c",
    "editorBracketPairGuide.activeBackground5": "#b48fad",
    "editorBracketPairGuide.activeBackground6": "#5f7083",
…
```

**Comment by cyqsimon** [comment]
I do not have much else to say except thank you for your work!

**Comment by Sun1090** [comment]
**Thank you for what you have done**！

**Comment by liusiyuxyfx** [comment]
Great Work, thank you!

**Comment by vibraniumdev** [comment]
Thank you for your amazing work through all these years <3 

Merry christmas! 🎄

**Comment by lclancey** [comment]
Thank you for the wonderful work on this project, I love this extension so much.
And Merry Christmas

**Comment by jjasoncool** [comment]
Thank you for your amazing work through all these years.
This extension helps me a lot!

Merry christmas! christmas_tree

**Comment by MuLinForest** [comment]
Thanks for your work.
Merry Christmas!

**Comment by umcai** [comment]
YYDS

**Comment by dingcang** [comment]
非常感谢。

**Comment by eynol** [comment]
Thanks for your work~  Merry christmas! 🎄

**Comment by CLDXiang** [comment]
Thank you for your amazing work!

**Comment by ticktockKK** [comment]
Thanks for your work~ Merry christmas! 🎄

**Comment by Vsnoy** [comment]
Thanks for your work！
Merry Christmas！🎄

**Comment by mpc2008cn** [comment]
Thanks for your work! Merry christmas! 🎄
My colors belows:
```
"workbench.colorCustomizations": {
        "editorBracketHighlight.foreground1": "#FFD700",
        "editorBracketHighlight.foreground2": "#DA70D6",
        "editorBracketHighlight.foreground3": "#87CEFA",
        "editorBracketHighlight.foreground4": "#FA8072",
        "editorBracketHighlight.foreground5": "#7CFC00",
        "editorBracketHighlight.foreground6": "#FF8C00"
      }
```

**Comment by Takazudo** [comment]
Thank you so much!!!
I just wanted to "thank YOU" awesome developers!

**Comment by 2hoyeong** [comment]
Thanks for your work over the last 5 years.

Merry christmas!

**Comment by GoudekettingRM** [comment]
Thanks for the extension and congrats on being so important you became native in VSCode 🚀

**Comment by qmisan** [comment]
Thank you for implementing this extension!

**Comment by debaraj-barua** [comment]
Thanks for the last 5 years!! ❤️

**Comment by stimw** [comment]
thanks for your great work!

btw, is anyone willing to share his colors for light themes (such as solarized light)?

**Comment by Nemvts** [comment]
Thank you for this amazing extension and all your hard work! I never knew I needed something like this in my dev life until now!

**Comment by jasikpark** [comment]
Thank you so much for making this! Best of luck with the future 💜

**Comment by LinkTsang** [comment]
Thanks for your work! ❤ 
Merry Christmas! 🎄

**Comment by hasban12138** [comment]
This is the best extension I have ever used
Thanks you

**Comment by ibakirov** [comment]
@CoenraadS Thanks for great work! I also congratulate you as your work become a native feature in VSCode!

**Comment by pujiaxun** [comment]
Thanks for your work! This extension is pretty cool! I used to recommend it to my colleagues~

**Comment by ayoh1** [comment]
thanks for work!

**Comment by jumoog** [comment]
Not sure if that possible but maybe you can add a button "migrate" and this will active everything?

**Comment by YM1099** [comment]
Thank you!
and Merry Christmas~🎄

**Comment by nattadasu** [comment]
Thank you for your hard work, and happy holiday! 😄

**Comment by yesIamFaded** [comment]
> ![image](https://user-images.githubusercontent.com/16220009/145954876-56debfa0-c3c8-4b36-b652-19fe87b82b6f.png)

How do you reach that View from inside VSCode? I dont know how to enable that, changing settings.json didnt work for me.

**Comment by SwapnilSoni1999** [comment]
Bow down to KING!

**Comment by ddlucad96** [comment]
Huge thanks for your work!

**Comment by Northsecond** [comment]
That's a Great work! Merry christmas!

**Comment by xfl12345** [comment]
@CoenraadS Thank you so much!!!  This extension really help me a lot!!! Thank you!!!❤💕

**Comment by Attackly** [comment]
Thanks for the Great work and making our work easier. Frohe Weihnachten 🎄❤

**Comment by Veerhan-glitch** [comment]
Thanks for the Great work and making our work easier
Merry Christmas！🎄

**Comment by TechStudent11** [comment]
I honestly think you should still maintain it. I still use it over the built-in one.

**Comment by icecream17** [comment]
Is there an equivalent for 
```json
"bracket-pair-colorizer-2.forceIterationColorCycle": true,
```
?

**Comment by Diogo-Rossi** [comment]
> That's not true at all. It supports the same custom colors as the extension and supports up to 6 unique levels, and it **does have horizontal bracket pair guides**.

@aminomancer, could you share the steps to put horizontal lines in bracket pairs, like in the [first gif  in my comment](https://github.com/CoenraadS/Bracket-Pair-Colorizer-2/issues/475#issuecomment-993511606)? 

I could not reproduce that, using exactly the same [settings you shared](https://github.com/CoenraadS/Bracket-Pair-Colorizer-2/issues/475#issuecomment-993674909). Fresh install of VS Code. No extensions.

![Animation1](https://user-images.githubusercontent.com/57204023/146841457-0ad09cca-9c1f-4ac9-a096-c2a71611fba1.gif)

I also tested changing all default `"active"` options to `"true"`, but it also didn't work.

**Comment by aminomancer** [comment]
Your gif isn't showing any condition where bracket pair guides would show. The guides only show when the brackets are on different lines. Horizontal setting just means the line traces the entire path from starting bracket to closing bracket, including the horizontal part.

**Comment by Diogo-Rossi** [comment]
>  The guides only show when the brackets are on different lines. Horizontal setting just means the line traces the entire path from starting bracket to closing bracket, including the horizontal part.

@aminomancer, thankyou for clarify.

So, this is a feature the native implementation does have yet: to put horizontal line in bracket pairs, when the brackets are on the same line, like this extension does. BTW, thankyou @CoenraadS, for this great work!

I hove Microsoft will include this feature in some release. Until then, I'll keep using this extension.

**Comment by aminomancer** [comment]
That's right, there's an [open issue](https://github.com/microsoft/vscode/issues/136475) requesting that enhancement

**Comment by erikahendsel** [comment]
Thank you so much for making everyone's code more colorful! 
Wish you Happy holidays!

**Comment by trinib** [comment]
this extension made people like coding.. goodbye

**Comment by icecream17** [comment]
For `"bracket-pair-colorizer-2.forceIterationColorCycle": true,` the vscode issues are https://github.com/microsoft/vscode/issues/131029 and https://github.com/microsoft/vscode/issues/131027

**Comment by waaronking** [comment]
> Like I said before, the only deficit in vs code's implementation is the inability to draw in the gutter

@rareyman What about doing additional custom styling? So far it looks like in vs-code `editorBracketMatch` only supports styling the `border` (color) and `background` (color). In BPC2 I can style both the `borderWidth` and `borderRadius`.

**Comment by Charles-Schleich** [comment]
Thank you for your work and for making many languages actually readable ! 
Many thanks and merry Christmas from 🇿🇦

**Comment by rareyman** [comment]
> @rareyman What about doing additional custom styling? So far it looks like in vs-code `editorBracketMatch` only supports styling the `border` (color) and `background` (color). In BPC2 I can style both the `borderWidth` and `borderRadius`.

Agreed. This awesome extension was able to do a lot more. Perhaps if we keep requesting these features to the VS Code team they can implement them…?

**Comment by lofairy** [comment]
Thank you for making my day better.

**Comment by XuehaiPan** [comment]
Thanks for making this great extension. I got lots of help while rewriting LaTeX manuscripts.

`Bracket-Pair-Colorizer` may not handle bracket pairs like `[x, y)` and `(x, y]` well with the latest version of `LaTeX-Workshop`:

![Screenshot brackets](https://user-images.githubusercontent.com/16078332/149658379-d1bfa3c8-0b36-4ead-94ab-3f586941b32f.png)

I have to transfer to the VSCode's builtin bracket colorization:

```jsonc
{
    //* Bracket Pair Colorization */
    "editor.guides.bracketPairs": true,
    "editor.guides.bracketPairsHorizontal": true,
    "editor.guides.highlightActiveBracketPair": true,
    "editor.guides.indentation": true,
    "editor.guides.highlightActiveIndentation": true,
    "editor.bracketPairColorization.enabled": true,
    "workbench.colorCustomizations": {
        "editorBracketHighlight.foreground1": "#ffd700",  // gold
        "editorBracketHighlight.foreground2": "#ff1491",  // deeppink
        "editorBracketHighlight.foreground3": "#00bfff",  // deepskyblue
        "editorBracketHighlight.foreground4": "#ff8c00",  // darkorange
        "editorBracketHighlight.foreground5": "#32cd32",  // limegreen
        "editorBracketHighlight.foreground6": "#ee82ee",  // violet
        "editorBracketHighlight.unexpectedBracket.foreground": "#ffffff"
    },
}
```

![Screenshot pastel](https://user-images.githubusercontent.com/16078332/149658535-6228e827-85d3-4360-8eb7-42323fcb2161.png)

I will miss the great feature of _**"show brackets in gutter"**_ and _**"show brackets in ruler"**_ which the built-in colorizer doesn't have.

**Comment by ryanxingql** [comment]
@CoenraadS Great work! Thank you!

**Comment by COD3BREAK3R** [comment]
Extension not work in VS Code 1.69.0

**Comment by Veerhan-glitch** [comment]
> Extension not work in VS Code 1.69.0

Dude it is no longer maintained. Read the issue u are posting in
